maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/service/FeatureDecisionServiceTest.java

```diff
@@ -36,18 +46,24 @@ void givenUserDoesNotHaveFeatureToggle_whenIsC3EnabledIsInvoked_thenReturnFalse(
             .build();
 
         FeatureDecisionService featureDecisionService = new FeatureDecisionService(maatCourtDataService);
-        boolean result = featureDecisionService.isC3Enabled(request);
+        Method method = FeatureDecisionService.class.getMethod(methodName, WorkflowRequest.class);
+        boolean result = (boolean) method.invoke(featureDecisionService, request);
```

Comment:
I'm not sure I'm too keen on the use of reflection here, I feel it would probably be easier to understand and maintain if we had separate test cases for each feature.

Reply:
This was always going to be a debatable change here :) I totally get the reasoning and I'm not a fan of magic either. The main reason I used reflection here is because the test methods for each of the individual methods are going to be the same except for the feature toggle and action, so with many methods we'd have a lot of duplication. The other benefit to the reflection approach is that we don't need to be constantly chopping and changing test methods as we add and remove the associated methods in the actual service; because the test conditions we're expecting are going to stay the same, we can simply update the methods names passed into the tests.

If we were to keep these tests as they are now, is there anything you can see which would help to make them easier to understand?